Chap_API_Server.tex

```diff
@@ -652,13 +652,21 @@ \subsubsection{Assembling the registration information}
 free(nodelist);
 
 /* pass the regex as the value to the PMIX_NODE_MAP key */
-PMIX_INFO_LOAD(&info, PMIX_NODE_MAP, regex, PMIX_STRING);
+PMIX_INFO_LOAD(&info, PMIX_NODE_MAP, regex, PMIX_REGEX);
 /* release the regex */
 free(regex);
 \end{codepar}
 \cspecificend
 
-Changing the filter criteria allows the construction of node maps for any level of information.
+Changing the filter criteria allows the construction of node maps for any level of information. The returned regular expression shall start with a string identifier of the method used to generate the expression. This is followed by a colon (':') delimiter and then the regular expression itself. The following identifiers are reserved by the Standard:
+
+\begin{itemize}
+    \item \emph{raw} - indicates that the expression is simply the comma-delimited input string (no processing was performed)
+    \item \emph{native} - a \ac{PMIx}-unique regular expression generator and parser
+    \item \emph{compress} - a compressed format based on the zlib compression package
+\end{itemize}
+
+Note that the regular expression itself is not required to be a printable string nor to obey typical string constraints (e.g., include a \code{NULL} terminator byte).
```

Comment:
We need to update the "Advice to PMIx server hosts" in both the `PMIx_generate_regex` and `PMIx_generate_ppn` functions (16.2.1 and 16.2.2). That advice is where we introduce the colon delimiter with either the `pmix:\0` prefix or the `blob:\0` prefix. Note that we added the `:\0` (with the NUL character) so that if the user tired to print it they wouldn't accidentally be printing an unterminated string. I think that it would be good to preserve that.

From this addition it seems like `native: == pmix:` and `compress: == blob:`. Why do we need the name change?

I think this text should be moved to `PMIx_generate_regex` (with a reference to it in `PMIx_generate_ppn`) as normal text since it is putting a requirement on those two functions. Then trim up the Advice blocks since some of that is normative text now.

In this particular section, it would be good to keep a sentence or two about the regex nature of the returned "string" specifically indicating that the return value is not necessarily printable (maybe keep your last sentence here).

Reply:
Sigh - just shows you shouldn't try to write documents when half asleep and multi-tasking on a telecon 😄 I just _knew_ I had written the prefix stuff up somewhere, but couldn't find it!

I'll do as suggested and push the change a little later - after I have completely woken up this time!

Reply:
@jjhursey Okay, see what you think now!